Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Location/LocationDrawingsTable.tsx

```diff
@@ -0,0 +1,191 @@
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  11/06/2024 - Collins Self
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import { SystemCenter } from "@gpa-gemstone/application-typings";
+import { Pencil, TrashCan } from "@gpa-gemstone/gpa-symbols";
+import { GenericController, LoadingScreen, ServerErrorIcon } from "@gpa-gemstone/react-interactive";
+import { ReactTable, Paging } from "@gpa-gemstone/react-table";
+import React from "react";
+import { useAppSelector } from "../hooks";
+import { SelectRoles } from "../Store/UserSettings";
+
+interface IProps {
+    LocationID: number,
+    Edit?: (record: SystemCenter.Types.LocationDrawing) => void,
+    /**
+     * @param UpdateTable Counter that triggers a fetchDrawings function
+     */
+    UpdateTable: number
+}
+
+const LocationDrawingsTable = (props: IProps) => {
+    const [links, setLinks] = React.useState<SystemCenter.Types.LocationDrawing[]>([]);
+    const [sortKey, setSortKey] = React.useState<keyof SystemCenter.Types.LocationDrawing>('Name');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [pageInfo, setPageInfo] = React.useState<{ RecordsPerPage: number, NumberOfPages: number, TotalRecords: number }>({ RecordsPerPage: 0, NumberOfPages: 0, TotalRecords: 0 });
+    const [pageState, setPageState] = React.useState<'error' | 'idle' | 'loading'>('idle');
+    const [page, setPage] = React.useState<number>(0);
+
+    const roles = useAppSelector(SelectRoles); // Deprecated
+    const LocationDrawingController = new GenericController<SystemCenter.Types.LocationDrawing>(`${homePath}api/LocationDrawing`, "Name", true);
+    const PagingID = 'LocationDrawingPage'
+
+    function hasPermissions(): boolean {
+        if (roles.indexOf('Administrator') < 0 && roles.indexOf('Engineer') < 0)
+            return false;
+        return true;
+    }
+
+    const fetchDrawings = (sortKey: keyof SystemCenter.Types.LocationDrawing, ascending: boolean, page: number, locationID: number) => {
+        setPageState('loading');
+        const handle = LocationDrawingController.PagedSearch([], sortKey, ascending, page, locationID)
+            .done((result) => {
+                setLinks(JSON.parse(result.Data as unknown as string));
```

Comment:
not confident this typing is need. what type is `LocationDrawingController.PagedSearch`?

Reply:
The type `LocationDrawingsController.PagedSearch` is returning is:  
`IPagedResult<SystemCenter.Types.LocationDrawing>`
and the `result.Data` is:
` SystemCenter.Types.LocationDrawing[]`

`setLinks()` does accept this data type, but without parsing it, I saw, `links` just gets set to the `results.Data` as a `string` of the array `results.Data`.
And within the `parse()` this typing was the only was I saw that would work. Casting the data with `JSON.stringify` also does not work when passing it to `JSON.parse()`

Reply:
This is a known issue, see
https://gridprotectionalliance.atlassian.net/browse/GEM-23